### PR TITLE
Clean up disabled tests #2

### PR DESCRIPTION
**Pull request by jsafrane**
* All tests that need specific cloud have proper Skip()
* Remove rules that don't match anything
* Enable tests that depend on 1.13 kubelet

This is continuation of https://github.com/openshift/origin/pull/22460

> @smarterclayton: This looks like it substantially increased test runtime.

I don't see any increase (`openshift/conformance/parallel` was actually slower before the PR than it is now, I blame low nr. of samples).

There are two new real tests:

```
"[sig-storage] GCP Volumes GlusterFS should be mountable [Suite:openshift/conformance/parallel] [Suite:k8s]"
"[sig-storage] PersistentVolumes NFS when invoking the Recycle reclaim policy should test that a PV becomes Available and is clean after the PVC is deleted. 
[Suite:openshift/conformance/parallel] [Suite:k8s]"
```

Additional 37 tests are skipped (missing `vsphere` / `openstack`), each `Skip()` takes 2-3 seconds for `BeforeEach` + `AfterEach` (divide by nr. of parallel tests).

cc @openshift/storage

### CODE REVIEW

**Comment by jsafrane** [comment]
Compared to a random green pull request #22898:

* e2e-aws with this PR: 1h44m (1h52m in #22898)
* e2e-aws-serial with this PR: 2h15m (2h12m in #22898)

/test e2e-aws
/test e2e-aws-serial
(to get more numbers)

**Comment by gnufied** [comment]
@jsafrane mostly looks good to me. I am skeptical about re-enabling ceph-rbd tests. They do have `Feature:Volumes` tag but otherwise there is no logic to skip them currently. And since we still haven't sorted out installation of ceph-common, they will be broken.

**Comment by jsafrane** [comment]
>  I am skeptical about re-enabling ceph-rbd tests. 

I think we should not enable them until we have all the binaries in place.

I don't want to do any smart detection in the tests - they should clearly fail when a binary is missing, so we get alert when someone removes them from our CoreOS images instead of silently skipping them.

File: test/extended/util/test.go [hunk 1]
```diff
@@ -351,12 +351,6 @@ var (
 			`should check if Kubernetes master services is included in cluster-info`, // Don't run kube-dns
 			`DNS configMap`, // this tests dns federation configuration via configmap, which we don't support yet
 
-			// vSphere tests can be skipped generally
-			`vsphere`,
-			`Cinder`, // requires an OpenStack cluster
-			// See the CanSupport implementation in upstream to determine wether these work.
-			`Ceph RBD`,                              // Works if ceph-common Binary installed (but we can't guarantee this on all clusters).
```
Comment (jsafrane) [inline test/extended/util/test.go]: Note to reviewers, this **does not** enable ceph tests! This rule does not match any test, ceph tests have something like `[Driver: rbd][Feature: Volumes]`.

**Comment by jsafrane** [comment]
/assign @smarterclayton 
for approval

**Comment by smarterclayton** [comment]
Only two new tests got run.  Is that expected?  (Master has 846, this has 848)

**Comment by jsafrane** [comment]
Yes, it is expected. Most of this PR is removing useless lines that don't match anything.

**Comment by jsafrane** [comment]
@smarterclayton now the tests are clean

**Pull request merged**